lib/msf/core/exploit/cmdstager.rb

```diff
@@ -55,6 +55,13 @@ def initialize(info = {})
     flavors = STAGERS.keys if flavors.empty?
     flavors.unshift('auto')
 
+    server_conditions = ['OPTION(CMDSTAGER::FLAVOR)', 'in', %w{certutil tftp wget curl fetch lwprequest psh_invokewebrequest}]
```

Comment:
Instead of having the data format be the end-user API, having objects and methods that represent the conditions that can then can then be serialized and evaluated can make using and implementing the rule evaluation much easier. For simple conditions there is not too much of a difference, but when using compound conditionals the difference is more pronounced. Then API could look something like this:

```suggestion
    server_conditions = OptCondition.new('CMDSTAGER::FLAVOR').in(*%w(certutil tftp wget curl fetch lwprequest psh_invokewebrequest))
```

Reply:
Thank you very much! The implementation of objects and methods to represent the conditions is a great idea. I'll write it down and work on it for a future improvement.